projects/coordinator/src/main/java/org/batfish/coordinator/resources/AuthenticationFilter.java

```diff
@@ -0,0 +1,65 @@
+package org.batfish.coordinator.resources;
+
+import java.security.AccessControlException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.ext.Provider;
+import org.batfish.common.CoordConsts;
+import org.batfish.common.Version;
+import org.batfish.coordinator.Main;
+
+/**
+ * This filter verify the access permissions for a user based on apiKey and version provided in
+ * request header.
+ */
+@Provider public class AuthenticationFilter implements ContainerRequestFilter {
+  @Override public void filter(ContainerRequestContext requestContext) {
+    String apiKey = CoordConsts.DEFAULT_API_KEY;
+    if (requestContext.getHeaders().containsKey(CoordConsts.SVC_KEY_API_KEY)) {
+      apiKey = requestContext.getHeaderString(CoordConsts.SVC_KEY_API_KEY);
+    }
+
+    String clientVersion = requestContext.getHeaderString(CoordConsts.SVC_KEY_VERSION);
+
+    // TODO: provides default version? Assume right version now
```

Comment:
For now, please separate `VersionFilter` from `AuthenticationFilter` -- Version and API Key checks should be done separately.

(If we see this is a performance bottleneck, we can re-combine.. but I doubt it for now).